app/server/models/user.py

```diff
@@ -260,20 +260,18 @@ def location(self):
 
     @location.setter
     def location(self, location):
+        from server.utils.location import async_set_user_gps_from_location
 
         self._location = location
 
         if location is not None and location is not '':
-
-            if self.id is None:
-                raise AttributeError('User ID not set')
-
-            try:
-                task = {'user_id': self.id, 'address': location}
-                task_runner('worker.celery_tasks.geolocate_address', args=(task,))
-            except Exception as e:
-                sentry_sdk.capture_exception(e)
-                pass
+            # Delay execution until after request to avoid race condition with db
+            # We still need to flush to get user id though
+            db.session.flush()
+            add_after_request_executor_job(
+                async_set_user_gps_from_location,
+                [self.id, location]
```

Comment:
Nit: Do you think we should use favour `kwargs` over `args` when using `add_after_request_executor_job` as a best practice? Just think it might be a bit more readable to see exactly what these arguments are in-line. I.e.   
`{'user_id' = self.id, 'location' = location}` explains something about the function signature without having to look up the actual`async_set_user_gps_from_location` function.

Reply:
yeah  totally fair, will do.